haunt-roll-fail/arcs/game-common.scala

```diff
@@ -781,9 +784,15 @@ object CommonExpansion extends Expansion {
 
         case BattleSystemAction(f, x, r, then) =>
             Ask(f).group(f, "battles in", r, x)
-                .each(f.rivals.%(_.present(r)))(e => BattleFactionAction(f, x, r, e, then).as(e))
+                .each(f.rivals.%(_.present(r)))(e => BattleHitBeforeAction(f, r, e, BattleFactionAction(f, x, r, e, BattleRepairAfterAction(f, r, e, then))).as(e))
```

Comment:
I can't exactly explain why I feel this way, but I think this nesting here is premature.

Also, I think we need to tunnel `used : $[Effect]` from the start all the way through battle actions.
One reason would be `Signal Breaker`/`Railgun Arrays` interaction - we could check and add `Signal Breaker` to `used` in this `PreBattleAction`, then process `Railgun Arrays`, then use `Signal Breaker` after roll if it was available.